test/smoke/tendermint.js

```diff
@@ -5,116 +5,134 @@ const getNetworkConfig = require('../../lib/test/getNetworkConfig');
 const { variables, inventory, network } = getNetworkConfig();
 
 describe('Tendermint', () => {
-  const masternodeStatuses = {};
+  const masternodeStatus = {};
+  const masternodeStatusError = {};
+  const masternodeStatusNodeInfo = {};
+  const masternodeStatusNetworkName = {};
 
-  for (const hostName of inventory.masternodes.hosts) {
-    describe(hostName, () => {
-      it('should be connected to the network', async function it() {
-        if (!variables.evo_services) {
-          this.skip('Evolution services are not enabled');
+  const masternodeNetInfo = {};
+  const masternodeNetInfoError = {};
 
-          return;
-        }
+  const blockHashes = {};
 
+  describe('All nodes', () => {
+    before('Collect tenderdash info', () => {
+      const promises = [];
+      for (const hostName of inventory.masternodes.hosts) {
         const tendermintClient = jaysonClient.http({
           // eslint-disable-next-line no-underscore-dangle
           host: inventory._meta.hostvars[hostName].public_ip,
           port: variables.tendermint_rpc_port,
         });
 
-        const response = await tendermintClient.request('status', {});
-
-        masternodeStatuses[hostName] = response;
-
-        const { result: { node_info: nodeInfo }, error } = response;
-
-        expect(error).to.be.undefined();
-
-        let networkName = `dash-${network.name}`;
-        if (network.type === 'devnet' && variables.tenderdash_chain_id !== undefined) {
-          networkName = `dash-${variables.tenderdash_chain_id}`;
-        }
-
-        expect(nodeInfo).to.deep.include({
-          network: networkName,
-          moniker: hostName,
-        });
-      });
-
-      it('should be connected to peers', async function it() {
-        if (!variables.evo_services) {
-          this.skip('Evolution services are not enabled');
+        const requestTendermintStatusPromise = tendermintClient.request('status', {})
+          .then((response) => {
+            masternodeStatus[hostName] = response;
+
+            const { result: { node_info: nodeInfo }, error } = response;
+            masternodeStatusError[hostName] = error;
+            masternodeStatusNodeInfo[hostName] = nodeInfo;
+
+            let networkName = `dash-${network.name}`;
+            if (network.type === 'devnet' && variables.tenderdash_chain_id !== undefined) {
+              networkName = `dash-${variables.tenderdash_chain_id}`;
+            }
+
+            masternodeStatusNetworkName[hostName] = networkName;
+          })
+          .catch((e) => {
+            // eslint-disable-next-line no-console
+            console.log(e);
```

Comment:
Won't it be better if you keep this error and pass to fail function below?

Reply:
Yes, I agree! But I couldn't figure out how to do this, because we already collect errors on line 33. I only added this because without catching errors and doing something (like `console.log()`) in the `.catch()` statement, the entire smoke test fails if a single node fails. I can't figure out why, I guess the error is not included in the response if the node isn't listening? I don't really understand how networking errors work or which module is actually throwing the error here, where can I learn about this stuff? How can I check network responses and see the status of a promise in the debugger?